### ISSUE TITLE
Eventhub Namespace Keeps Resetting Capacity

### ISSUE THREAD

**Issue by CodingAnarchy** [terraform-providers/terraform-provider-azurerm#6929]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.12.24
+ provider.azurerm v2.9.0

### Affected Resource(s)

* `azurerm_eventhub_namespace`

### Terraform Configuration Files

```hcl
resource "azurerm_eventhub_namespace" "foo-namespace" {
  name = "foo-namespace"
  location = azurerm_resource_group.resources.location
  resource_group_name = azurerm_resource_group.resources.name
  sku = "Standard"
  auto_inflate_enabled = true
  maximum_throughput_units = 20
}
```

### Expected Behavior

State hasn't changed, so there should be no queued changes for the Event Hub Namespace.

### Actual Behavior

```
 ~ resource "azurerm_eventhub_namespace" "foo-namespace" {
        auto_inflate_enabled                = true
      ~ capacity                            = 2 -> 1
        default_primary_connection_string   = (sensitive value)
        default_primary_key                 = (sensitive value)
        default_secondary_connection_string = (sensitive value)
        default_secondary_key               = (sensitive value)
        id                                  = "/subscriptions/[subscription id]/resourceGroups/resources/providers/Microsoft.EventHub/namespaces/foo-namespace"
        location                            = "eastus2"
        maximum_throughput_units            = 20
        name                                = "foo-namespace"
        network_rulesets                    = [
            {
                default_action       = "Deny"
                ip_rule              = []
                virtual_network_rule = []
            },
        ]
        resource_group_name                 = "resources"
        sku                                 = "Standard"
        tags                                = {}
    }
```

### Steps to Reproduce

1. `terraform plan`

**Comment by neil-yechenwei** [comment]
Thanks for opening this issue. `Capacity` has default value 1. After tested, seems I cannot repro the issue with below tfconfig. Maybe I missed something. If you need, you can try below tf config to check whether the issue still exists.

tfconfig:
```
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "eventhubnsrg"
  location = "westus"
}

resource "azurerm_eventhub_namespace" "test" {
  name                     = "foo-namespacetest"
  location                 = azurerm_resource_group.test.location
  resource_group_name      = azurerm_resource_group.test.name
  sku                      = "Standard"
  auto_inflate_enabled     = true
  maximum_throughput_units = 20
}
```

Result:
```
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

azurerm_resource_group.test: Refreshing state... [id=/subscriptions/xx-xx-xx-xx/resourceGroups/eventhubnsrg]
azurerm_eventhub_namespace.test: Refreshing state... [id=/subscriptions/xx-xx-xx-xx/resourceGroups/eventhubnsrg/providers/Microsoft.EventHub/namespaces/foo-namespacetest]

------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.
```

**Comment by CodingAnarchy** [comment]
@neil-yechenwei I suspect that you could not replicate in a test because, although auto-inflate was enabled, it was not actually used. In our case, auto-inflate will adjust the capacity up to handle larger amounts of incoming traffic, but then terraform will reset the capacity to 1 when refreshing and reapplying the state.

We need it to allow, when auto-inflate is enabled, the existing status in Azure to override the default capacity.

**Comment by neil-yechenwei** [comment]
Thanks for clarification. Seems I got your point. Per my understanding, terraform cannot know capacity has been changed after first apply since capacity would dynamically increase while incoming traffic increases. And capacity could be modified so that we cannot suppress the diff from terraform side. So I think it's by design and suggest you to use ignore_changes to avoid this problem. Thanks.
```
resource "azurerm_eventhub_namespace" "test" {
  .......
  
  lifecycle {
    ignore_changes = [capacity]
  }
}
```

Related reference: https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-auto-inflate
